src/test/java/se/sundsvall/messaging/integration/db/StatisticsRepositoryByDepartmentTest.java

```diff
@@ -0,0 +1,70 @@
+package se.sundsvall.messaging.integration.db;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.transaction.annotation.Transactional;
+import se.sundsvall.messaging.integration.db.projection.StatsEntry;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.groups.Tuple.tuple;
+import static se.sundsvall.messaging.model.MessageStatus.FAILED;
+import static se.sundsvall.messaging.model.MessageStatus.SENT;
+import static se.sundsvall.messaging.model.MessageType.LETTER;
+import static se.sundsvall.messaging.model.MessageType.SNAIL_MAIL;
+
+@SpringBootTest
+@ActiveProfiles("junit")
+@Sql(scripts = {
+	"/db/scripts/truncate.sql",
+	"/db/scripts/testdata-junit.sql"
+})
+@Transactional
+class StatisticsRepositoryByDepartmentTest {
+
+	@Autowired
+	private StatisticsRepository statisticsRepository;
+
+
+	@Test
+	void getStatsByDepartment() {
+
+		final var letterStatEntries = statisticsRepository.getStatsByDepartment("SBK(Gatuavdelningen, Trafiksektionen)", LETTER, null, null);
+
+
+		// Assert that the map contains the expected keys
+		assertThat(letterStatEntries).isNotNull().isNotEmpty().hasSize(4)
+			.extracting(StatsEntry::department, StatsEntry::messageType, StatsEntry::originalMessageType, StatsEntry::status)
+			.containsExactly(
+				tuple("SBK(Gatuavdelningen, Trafiksektionen)", SNAIL_MAIL, LETTER, SENT),
+				tuple("SBK(Gatuavdelningen, Trafiksektionen)", SNAIL_MAIL, LETTER, SENT),
+				tuple("SBK(Gatuavdelningen, Trafiksektionen)", SNAIL_MAIL, LETTER, SENT),
+				tuple("SBK(Gatuavdelningen, Trafiksektionen)", SNAIL_MAIL, LETTER, FAILED));
+	}
+
+	@Test
+	void getStatsByDepartmentNoDepartment() {
+
+		final var letterStatEntries = statisticsRepository.getStatsByDepartment(null, LETTER, null, null);
+
+
+		// Assert that the map contains the expected keys
+		assertThat(letterStatEntries).isNotNull().isNotEmpty().hasSize(11)
```

Comment:
No big deal, but "isNotNull().isNotEmpty().hasSize(11)" is redundant since you also have an "containsExactlyInAnyOrder" below.